sklearn/utils/_typedefs.pyx

```diff
@@ -1,20 +1,19 @@
-#!python
+# _typedefs is a declaration only module
+# The functions implemented here are for testing purpose.
```

Comment:
I think we should rename this `_typedefs.pyx` file to `_typedefs_testing.pyx` or `_typedefs_test_helper.pyx` (and update the build setup and test accordingly) to avoid any confusion. In particular I wouldn't want contributors to be confused as to when to se the `type_t` fused type (the answer should be "never").